File: backend-python/app/models/medication.py
from sqlalchemy import Column, Integer, String, Date, Boolean, Text
from app.database import Base

class Medication(Base):
    __tablename__ = "medications"
    
    id = Column(Integer, primary_key=True, index=True)
    pet_id = Column(Integer, nullable=False, index=True)
    name = Column(String, nullable=False)
    dosage = Column(String, nullable=False)
    quantity_remaining = Column(Integer)
    refill_date = Column(Date)
    medication_type = Column(String)
    purpose = Column(String)
    is_active = Column(Boolean, default=True)
    notes = Column(Text)

File: backend/src/main/java/com/petcare/model/Medication.java
package com.petcare.model;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDate;

@Entity
@Table(name = "medications")
@Data
public class Medication {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(name = "pet_id", nullable = false)
    private Long petId;
    
    @Column(nullable = false)
    private String name;
    
    @Column(nullable = false)
    private String dosage;
    
    @Column(name = "quantity_remaining")
    private Integer quantityRemaining;
    
    @Column(name = "refill_date")
    private LocalDate refillDate;
    
    @Column(name = "medication_type")
    private String medicationType;
    
    private String purpose;
    
    @Column(name = "is_active")
    private Boolean isActive = true;
    
    @Column(columnDefinition = "TEXT")
    private String notes;
}
